core/drand.go

```diff
@@ -18,15 +19,17 @@ import (
 // can start the DKG, read/write shars to files and can initiate/respond to TBlS
 // signature requests.
 type Drand struct {
+	ctx  context.Context
```

Comment:
Why having a long lived context ? Context are generally for like a request or set of requests, that you pass down in all layers of the call stack, not long lived structures to keep around.

Reply:
The ctx is needed to register some of the GRPC listeners (RegisterPublicHandlerClient). What is wrong with a long lived context?

Reply:
The context is needed here: https://github.com/drand/drand/blob/master/protobuf/drand/api.pb.gw.go#L275
(This is the auto-generated code.)
In general, from what I can tell, there is nothing precluding you from having long-lived context. And in this case it seems that the GRPC framework requires it?

Reply:
[Drive-by]: nothing technically precludes it, but it's almost always the wrong thing to do. Do the plumbing if at all possible.

https://golang.org/pkg/context/
> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it.

Reply:
@nikkolasg Even though RegisterPublicHandlerClient requires a context argument, it doesn't seem like it ever uses it. How was api.pb.gw.go generated? It seems like we could pass anything (even nil) for context, and it would still work.